data/abilities.ts

```diff
@@ -4670,8 +4670,8 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 		num: 309,
 	},
 	terashell: {
-		// TODO figure out if this only works on Terapagos
 		onEffectiveness(typeMod, target, type, move) {
+			if (!target.species.id === 'terapagosterastal') return;
 			if (this.effectState.resisted) return -1; // all hits of multi-hit move should be not very effective
 			if (!target || move.category === 'Status') return;
```

Comment:
```suggestion
			if (!target || pokemon.baseSpecies.baseSpecies !== 'Terapagos-Terastal') return;
			if (this.effectState.resisted) return -1; // all hits of multi-hit move should be not very effective
			if (move.category === 'Status') return;
```
Make this change and it will be good to go :)

Reply:
It should be baseSpecies.name not baseSpecies.baseSpecies and target instead of pokemon

Reply:
Tera Shift: `if (pokemon.baseSpecies.baseSpecies !== 'Terapagos' || pokemon.transformed) return;`
https://github.com/smogon/pokemon-showdown/blob/494919db160f0fdc2e323ac04e4a3f52e2ef6ae8/data/abilities.ts#L4694

Thick Club: `if (pokemon.baseSpecies.baseSpecies === 'Cubone' || pokemon.baseSpecies.baseSpecies === 'Marowak') {`
https://github.com/smogon/pokemon-showdown/blob/494919db160f0fdc2e323ac04e4a3f52e2ef6ae8/data/items.ts#L5804

Reply:
Thick club specifically uses baseSpecies because it needs to work on the Kanto and Alolan formes. Remember that baseSpecies returns the mon without the forme name, so here it will always return Terapagos and not Terapagos-Terastal, which is why I said to use name (which will include the forme name)

Reply:
Thank you, changed!